src/electron/vpn_tunnel.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export interface VpnTunnel {
+  // Starts a system-wide VPN that tunnels IP traffic.
+  connect(): Promise<void>;
+
+  // Stops tunneling and tears down the VPN.
+  disconnect(): Promise<void>;
+
+  // Resolved once the tunnel is disconnected after `disconnect` is called,
+  // or when the tunnel disconnects spontaneously.
+  readonly onceDisconnected: Promise<void>;
+
+  // Sets an optional callback for when the tunnel is attempting to re-connect.
+  onReconnecting(listener?: (() => void)): void;
```

Comment:
Perhaps not for this PR, but a more mainstream way to do this would be to expose on/once methods, and use a EventEmitter:
https://nodejs.org/api/events.html#events_events

And onceDisconnected should be a 'disconnected' event. The called can install a once() handler.

Reply:
Ack, should I add a TODO?

Reply:
A TODO sounds good.

Reply:
Done in 18e9db6.